### DISCUSSION TITLE
2 way audio works, and then stops working

### DISCUSSION THREAD

**Discussion by leaskovski** [blakeblackshear/frigate#14066] [General Support] [answered]
Hello all...  I have followed the docs, and have been able to get 2 way audio to work with my tapo c200 camera.  Im using my Home Assistant setup to access frigate, so that HTTPS access to frigate is made via that, to allow for the browser to ask for access to the microphone.

This all works...  well to start with.  I start up my frigate container on my VM, go to HA, open up the frigate proxy addon to see frigate, go to my tapo c200 camera, and then use the microphone to start the two way audio and speak... all is good and I can hear myself from the other room.  I then close the page down to HA and wait.  About 5 minutes later, I go back to HA, open up the frigate proxy addon and view frigate, go to my camera and repeat the same test, but this time now audio.

How do I even debug this?  I am assuming its a go2rtc issue, so I have enabled "log: level: debug" in the go2rtc section of the config and restarted.  Looking through the logs though, there isnt anything that is helping me.  What log levels do I really need for this?

**Comment by hawkeye217** [comment]
Sounds like it might be something related to go2rtc. You could try using two way talk through the go2rtc dashboard to rule out something with HA.

**Reply by leaskovski** [reply]
Thanks for your reply.  Im being dump though... how do I get to the go2rtc dashboard?  Remember that for a browser to be able to use a mic it has to be a https site, and my frigate setup isn't setup for https... thats done using HA and proxying to frigate.

**Reply by hawkeye217** [reply]
That's correct, a secure context is required. If you are using a proxy, you can map port 1984 through.

**Reply by leaskovski** [reply]
Just looking at the docs, it looks like my docker needs to have port 1984 opened up, which isn't included in with the docker run command by default, so looks like I need to faff with my container first

**Reply by leaskovski** [reply]
I am being proper stupid...  Im sure there used to be a button to open the go2rtc dashboard from within the Frigate config pages...  Has that been removed now?  I have opened the port, so I can connect to the dashboard now via any internal browser.  Im now going to have to hack about with my nginx setup to effectively proxy the internal machine up to have https.  Im sure this used to be easier in earlier versions... I don't remember ever having to do this to test my go2rtc module.

**Reply by NickM-27** [reply]
it was removed due to security issues, however there is no need for https specifically as technically if you are using chrome you can use a flag to treat a specific page as secure

**Reply by leaskovski** [reply, marked as answer]
Thank you!  I managed to use the go2rtc dashboard in edge (don't have chrome installed, and firefox doesnt have such a feature), and get the 2 way audio webrtc link working.  Tested and it work.  Closed the tab, and then waited for a while and reopened and then the audio wasn't working.  Looks like I will need to open up an issue on the go2rtc github page.
